lib/kafka/consumer.rb

```diff
@@ -307,6 +307,17 @@ def poll(min_bytes: 1, max_wait_time: 5)
       batches = fetch_batches(min_bytes: min_bytes, max_wait_time: max_wait_time)
       send_heartbeat_if_necessary
       batches
+    rescue HeartbeatError, OffsetCommitError
+      join_group
+      retry
+    rescue FetchError, NotLeaderForPartition, UnknownTopicOrPartition
+      @cluster.mark_as_stale!
+      retry
+    rescue LeaderNotAvailable => e
+      @logger.error "Leader not available; waiting 1s before retrying"
+      @cluster.mark_as_stale!
+      sleep 1
+      retry
```

Comment:
Do we want to retry indefinitely?

Reply:
that's a good question. the reason i was thinking to retry is to let the consumer rebalance while they're trying to poll. But maybe having a max number of retries would be a better approach. What do you think about 10 retries?